apps/api/src/controllers/competition.controller.ts

```diff
@@ -479,69 +483,66 @@ export function makeCompetitionController(services: ServiceRegistry) {
         const { competitions, total } =
           await services.competitionManager.getCompetitions(
             status,
+            // Default limit 10, max 100. It's important we don't call this without a limit
             pagingParams,
           );
 
         // If user is authenticated, enrich competitions with voting information
         let enrichedCompetitions = competitions;
         if (userId) {
-          enrichedCompetitions = await Promise.all(
-            competitions.map(async (competition) => {
```

Comment:
This `await Promise.all( ... array.map(...` with database calls inside it is one of the obvious patterns we can search for to find places the db will get overloaded when data gets bigger.  This PR removes this and simply does two queries that involve joins.  reducing the max query count for this controller from 402 down to always being 3.